tldextract/tldextract.py

```diff
@@ -187,8 +187,15 @@ def __call__(self, url):
             .rstrip(".")
 
         is_punycode = netloc.startswith('xn--') or '.xn--' in netloc
+        # Some IDN urls might generate an error
         if is_punycode:
-            netloc = codecs.decode(netloc.encode('ascii'), 'idna')
+            try:
+                netloc = codecs.decode(netloc.encode('ascii'), 'idna')
+            except UnicodeError:
+                LOG.warn('UnicodeError when decoding netloc ' + netloc)
```

Comment:
Cool!

I understand where you're coming from. Outside this library, I don't like failing silently either. I like preventing errors before they can even occur.

But it's this library's stance that, there's plenty of URL validators out there. Rather than inflict their overhead on all users, if it's important, they can opt into validation themselves. And it seems to be Python's idiom to avoid typechecks, be adults, RTFM (often, RTFS), and pass expected args. Keeps this library light and easy to get started with. I will document this design approach better.

I do wonder if this lenient approach has desensitized users to the nuances of URLs in the wild...